src/main/java/com/google/cloud/tools/appengine/v2/deploy/DeployRequest.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.v2.deploy;
+
+import com.google.cloud.tools.appengine.v2.OutputHandler;
+
+import java.util.concurrent.Future;
+
+/**
+ * Public API for Deploy Requests.
+ */
+public interface DeployRequest {
```

Comment:
I don't see this class being used (just referred by a javadoc comment in DeployResult). Should it inherit from `CloudSdkRequest` and be the return value of `AppEngineRequestFactory.newDeploymentRequest(...)`?